controller/controller.go

```diff
@@ -0,0 +1,197 @@
+	r.Get("/images/branding/product/2x/translate_24dp.png", GetGStaticResource)
+
+	return r
+}
+
+func getHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout: time.Second * 10,
+	}
+}
+
+// GetLanguageList send a request to Microsoft server and convert the response
+// into google format and reply back to the client.
+func GetLanguageList(w http.ResponseWriter, r *http.Request) {
+	// Send a get language list request to MS
+	req, err := http.NewRequest("GET", MSTranslateServer+languageEndpoint, nil)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error creating MS request: %v", err), http.StatusInternalServerError)
+	}
+
+	client := getHTTPClient()
+	msResp, err := client.Do(req)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error sending request to MS server: %v", err), http.StatusInternalServerError)
+		return
+	}
+	defer func() {
+		err := msResp.Body.Close()
+		if err != nil {
+			log.Errorf("Error closing response body stream: %v", err)
+		}
+	}()
+
+	// Set response header
+	w.Header().Set("Content-Type", msResp.Header["Content-Type"][0])
+	w.WriteHeader(msResp.StatusCode)
+
+	// Copy resonse body if status is not OK
+	if msResp.StatusCode != http.StatusOK {
+		_, err = io.Copy(w, msResp.Body)
+		if err != nil {
+			http.Error(w, fmt.Sprintf("Error copying MS response body: %v", err), http.StatusInternalServerError)
+		}
+		return
+	}
+
+	// Convert to google format language list and write it back
+	msBody, err := ioutil.ReadAll(msResp.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error reading MS response body: %v", err), http.StatusInternalServerError)
+	}
+	body, err := language.ToGoogleLanguageList(msBody)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error converting to google language list: %v", err), http.StatusInternalServerError)
+		return
+	}
+	_, err = w.Write(body)
+	if err != nil {
+		log.Errorf("Error writing response body for translate requests: %v", err)
+	}
+}
+
+// Translate converts a Google format translate request into a Microsoft format
+// one which will be send to the Microsoft server, and write a Google format
+// response back to the client.
+func Translate(w http.ResponseWriter, r *http.Request) {
+	// Convert google format request to MS format
+	req, isAuto, err := translate.ToMicrosoftRequest(r, MSTranslateServer)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error converting to MS request: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	// Send translate request to MS server
+	client := getHTTPClient()
+	msResp, err := client.Do(req)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Error sending request to MS server: %v", err), http.StatusInternalServerError)
+		return
+	}
+	defer func() {
+		err := msResp.Body.Close()
+		if err != nil {
+			log.Errorf("Error closing response body stream: %v", err)
+		}
+	}()
+
+	// Set Header
+	w.Header().Set("Content-Type", msResp.Header["Content-Type"][0])
+	w.Header().Set("Access-Control-Allow-Origin", "*") // same as Google response
```

Comment:
`https://translate.googleapis.com` instead of *?

Reply:
This is mimicking google response and they uses *, so I just follow them to avoid any surprises.